### PR TITLE
Fixing Branch Manager Handbook url in README.md

### PR DESCRIPTION
**Pull request by pswica**
Just a small documentation PR. 

I noticed the link for the handbook was missing when I was looking for it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ Tools | Description
 <br/> [`push-build.sh`](https://github.com/kubernetes/release/blob/master/push-build.sh) <br/><br/> | Pushes a developer build or CI Jenkins build up to GCS.
 <br/> [`script-template`](https://github.com/kubernetes/release/blob/master/script-template) <br/><br/> | Generates a script template in the kubernetes/release ecosystem.
 
-For information on how to use `gcbmgr`, `anago` and `branchff`, see the [Branch Manager Handbook](https://github.com/kubernetes/sig-release/tree/master/release-team/role-handbooks/branch-manager#branch-manager-handbook)
+For information on how to use `gcbmgr`, `anago` and `branchff`, see the [Branch Manager Handbook](https://github.com/kubernetes/sig-release/blob/master/release-engineering/role-handbooks/branch-manager.md)
```
Comment (justaugustus) [inline README.md]:
```suggestion
For information on how to use `gcbmgr`, `anago` and `branchff`, see the [Branch Manager Handbook](https://git.k8s.io/sig-release/release-engineering/role-handbooks/branch-manager.md)
```

**Comment by pswica** [comment]
Changes made. Thank you for reviewing!

**Pull request merged**